Controllers/messageController.js

```diff
@@ -1,13 +1,38 @@
 const CatchAsyncError = require('../Utils/CatchAsyncError')
 const db = require('../db/models/index')
 const Message = db.Message
+const Conversation = db.Conversation
+const User = db.User
 
 exports.create = CatchAsyncError(async (req, res, next) => {
   const message = await Message.create({
     userId: req.user.id,
     conversationId: req.params.id,
     body: req.body.text
   })
+  const conversation = await Conversation.findAll({
+    attributes: ['id'],
+    raw: true,
+    nest: true,
+    where: {
+      id: req.params.id
+    },
+    include: [
+      {
+        model: User,
+        attributes: ['id', 'username'],
+        through: {
+          attributes: []
+        }
+      }
+    ]
+  })
+
+  // Emiting socket events
+  conversation.forEach((conversation) => {
+    req.io.to(conversation.Users.username).emit('newMessage', message.dataValues)
+  })
+
```

Comment:
FIx the conversation query. You can write this as:
```js
  const conversation = await Conversation.findOne({
    where: {
      id: req.params.id
    },
    include: {
      model: User,
      attributes: ['id', 'username']
    }
  })
```
Avoid using the raw argument in the queries.
And since you're already querying for a conversation move this as the first thing in the controller. And have the message create use the conversation id from the result.
Also fix the emit loop. Should be conversation.users.forEach